tests/test_harvester_ocn_inc_logs.py

```diff
@@ -0,0 +1,139 @@
+    
+    stats_harvested = list()
+    pt_tuple_subset = list()
+    data1 = harvest(VALID_CONFIG_DICT)
+    for i, harvested_data_tuple in enumerate(data1):
+        if np.isin(i, valid_idxs_in_tuple): # pt_inc data
+            assert harvested_data_tuple[1] in VALID_CONFIG_DICT['statistic']
+            stats_harvested.append(harvested_data_tuple[1])
+            assert harvested_data_tuple[2] == VALID_CONFIG_DICT['variable'][variable_rank]
+            
+            pt_tuple_subset.append(harvested_data_tuple)
+    
+    for i, valid_statistic in enumerate(VALID_CONFIG_DICT['statistic']):
+        assert valid_statistic == stats_harvested[i]
+        assert valid_statistic == 'mean' or valid_statistic == 'RMS'
+        
+        if valid_statistic == 'mean':
+            assert pt_tuple_subset[i][3] == -0.657E-02
+        elif valid_statistic == 'RMS': # RMS
+            assert pt_tuple_subset[i][3] == 0.944E-01
+                        
+def test_s_inc():
+    harvest_data('s_inc', 0.108E-02, 0.330E-01)
+    
+def test_u_inc():
+    harvest_data('u_inc', 0.300E-03, 0.304E-01)
+    
+def test_v_inc():
+    harvest_data('v_inc', 0.155E-03, 0.283E-01)
+    
+def test_ssh():
+    harvest_data('SSH', 0.667E-01, 0.664E+00)
+    
+def test_salinity():
+    harvest_data('Salinity', 0.348E+02, 0.348E+02)
+    
+def test_temperature():
+    harvest_data('Temperature', 0.112E+02, 0.150E+02)
+    
+def test_speed_of_currents():
+    harvest_data('Speed of Currents', 0.928E-01, 0.154E+00)
+
+def harvest_data(varname, expected_ans_0, expected_ans_1):
+    for i, variable in enumerate(VALID_CONFIG_DICT['variable']):
+        if variable == varname:
+            variable_rank = i
+    assert VALID_CONFIG_DICT['variable'][variable_rank]==varname
+    assert VALID_CONFIG_DICT['variable'][variable_rank]==varname
+    
+    n_stats = len(VALID_CONFIG_DICT['statistic'])
+    
+    valid_idxs_in_tuple = np.arange(n_stats) + variable_rank * n_stats
+    
+    stats_harvested = list()
+    valid_tuple_subset = list()
+    data1 = harvest(VALID_CONFIG_DICT)
+    for i, harvested_data_tuple in enumerate(data1):
+        
+        if np.isin(i, valid_idxs_in_tuple):
+            assert harvested_data_tuple[1] in VALID_CONFIG_DICT['statistic']
+            stats_harvested.append(harvested_data_tuple[1])
+            assert harvested_data_tuple[2] == VALID_CONFIG_DICT['variable'][variable_rank]
+            
+            valid_tuple_subset.append(harvested_data_tuple)
+    
+    for i, valid_statistic in enumerate(VALID_CONFIG_DICT['statistic']):
+        assert valid_statistic == stats_harvested[i]
+        assert valid_statistic == 'mean' or valid_statistic == 'RMS'
+        
+        if valid_statistic == 'mean':
+            assert valid_tuple_subset[i][3] == expected_ans_0
+        elif valid_statistic == 'RMS': # RMS
+            assert valid_tuple_subset[i][3] == expected_ans_1
+
+def run_tests():
```

Comment:
this is unusual for a test suite, as is the main function. pytest can run the entire file of test cases without it. do you need to have it or can we remove it and use pytest as the other test files in the repo do?

Reply:
It isn't necessary. But pytest still works with this function in place, plus I can run the test suite in ipython. It's helpful for interactive debugging.